rest_framework_tracking/mixins.py

```diff
@@ -40,21 +40,17 @@ def initial(self, request, *args, **kwargs):
         else:
             view_method = method.lower()
 
-        # save to log
-        try:
-            self.request.log = APIRequestLog.objects.create(
-                requested_at=now(),
-                path=request.path,
-                view=view_name,
-                view_method=view_method,
-                remote_addr=ipaddr,
-                host=request.get_host(),
-                method=request.method,
-                query_params=_clean_data(request.query_params.dict()),
-            )
-        except Exception:
-            # tracking db constraints should not create a failure on the API
-            return
+        # create log
+        self.request.log = APIRequestLog(
```

Comment:
Well, yes, that's the point. But then, the object is carried over through request processing and we [do save in DB at the end](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L111).

With these changes we have only **one** call to the DB instead of:
* **3** for successful responses ([#L45](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L45), [#L78](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L78), [#L111](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L111))
* **4** for exceptions ([#L45](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L45), [#L78](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L78), [#L88](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L88), [#L111](https://github.com/aschn/drf-tracking/blob/master/rest_framework_tracking/mixins.py#L111))

I think that this kind of optimizations is not pointless, as we surely want to limit the overhead brought by the tracking.